src/components/molecules/MenuListItem/type.ts

```diff
@@ -0,0 +1,21 @@
+import { ReactElement, ReactNode } from "react";
+
+type TypeAndDescriptionType =
+  | {
+      type: "header";
+      description?: string;
+    }
+  | {
+      type: "dropdown";
+      description?: never;
+    };
+
+export type MenuListItemProps = {
+  chevron?: boolean;
+  disabled?: boolean;
+  icon?: ReactElement;
```

Comment:
I think I might prefer combining these types:
```suggestion
export type MenuListItemProps = {
  type: "header" | "dropdown";
  description?: string;
  chevron?: boolean;
  disabled?: boolean;
  icon?: ReactElement;
```
At least to me that would be easier to read. What do you think?